### ISSUE TITLE
psthRasterAndCounts gives psth that is undercounted in at the edge-bins

### ISSUE THREAD

**Issue by ilandau** [cortex-lab/spikes#12]
Hey there, I'm a new user. I noticed that the psth created by psthRasterAndCounts was undercounted in the first and last bin. Both _psth_ and _bins_ are returned as vectors that are the length of the number of **edges**, i.e. one more than the number of **bins**. This could be related to the old matlab _hist_ function which used to accept **bin-centers** as inputs, whereas the new _histogram_ function only accepts a vector of **edges** as inputs.

In any case, I made minor changes in the file locally to make it return _psth_ and _bins_ corresponding to the **bin-centers** and with length corresponding to the actual number of bins. I'd be happy to share that change. But maybe people out there are still using old versions of Matlab and that's why they're not running into issues?

**Comment by nsteinme** [comment]
Good catch, yes looks like the first bin will be half as wide as the others (this is only going to be true for the last bin if the window size is evenly divisible by the bin size - I don't think it applies to the last bin in general). I think the correct fix is to change line 31 to start from `window(1)+psthBinSize/2` rather than starting from `window(1)`. Do you agree?

**Comment by ilandau** [comment]
So what I did was to replace line 31 with the following:

`edges = window(1):psthBinSize:window(2);
[psth, edges] = histcounts(stRelToEvent, edges);
bins = edges(1:end-1) + psthBinSize/2; `

Matlab seems to highly recommend switching out the old calls to _hist_.

**Comment by nsteinme** [comment]
Ok, looks good, could you make a pull request with that?

**Comment by ilandau** [comment]
Sure.
